src/app/components/SubjectHeading/SubjectHeadingSearch.jsx

```diff
@@ -0,0 +1,135 @@
+import React from 'react';
+import { Link } from 'react-router';
+import PropTypes from 'prop-types';
+import axios from 'axios';
+import appConfig from '../../../../appConfig';
+import AutosuggestItem from './AutosuggestItem'
+
+class SubjectHeadingSearch extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      activeSuggestion: 0,
+      suggestions: [],
+      userInput: ''
+    }
+
+    this.onChange = this.onChange.bind(this)
+    this.onSubmit = this.onSubmit.bind(this)
+    this.clearUserInput = this.clearUserInput.bind(this)
+    this.generatePath = this.generatePath.bind(this)
```

Comment:
I think binding the reference is only necessary if we're passing `generatePath` to another component, which - depending on how you feel about a related comment - you may not wish to do